gensim/test/test_matutils.py

```diff
@@ -141,6 +142,43 @@ def testDirichletExpectation(self):
                 self.assertTrue(np.allclose(known_good, test_values), msg)
 
 
+class UnitvecTestCase(unittest.TestCase):
+    # test unitvec
+    def manual_unitvec(self, vec):
+        self.vec = vec
+        self.vec = self.vec.astype(np.float)
+        if sparse.issparse(self.vec):
+            vec_sum_of_squares = self.vec.multiply(self.vec)
+            unit = 1. / np.sqrt(vec_sum_of_squares.sum())
+            return self.vec.multiply(unit)
+        elif not sparse.issparse(self.vec):
+            sum_vec_squared = np.sum(self.vec ** 2)
+            self.vec /= np.sqrt(sum_vec_squared)
+            return self.vec
+
+    def test_inputs(self):
```

Comment:
IMO we should split this test into multiple tests (one per combination of `(arr_type, dtype)` maybe) so the logic is simpler and so that it's possible to look at a failing test log and see exactly what kind of input caused the test to fail.